### ISSUE TITLE
Poor colour rendering on Windows - with a TrueColor terminal

### ISSUE THREAD

**Issue by ZePhant** [browsh-org/browsh#404]
OS: Windows 11
Terminal: Windows Terminal

The Windows version of Browsh seems to be inferior to the Linux version for some reason. Windows Terminal is capable of rendering Truecolor colors, but Browsh **on Windows** doesn't render colors properly.

The same version of Browsh on Kali Linux, running on Windows Terminal seems to render colors correctly.

On Linux:
![image](https://user-images.githubusercontent.com/77280713/159147402-e0fcc809-5666-4088-917d-6034029b9f7f.png)
![image](https://user-images.githubusercontent.com/77280713/159147415-0aec8cdd-2243-4aad-afb8-4aefbf529c4b.png)

On Windows:

**Comment by tombh** [comment]
Have you followed the tests and advice on this gist? https://gist.github.com/XVilka/8346728

**Comment by ZePhant** [comment]
The terminal might not be the problem @tombh , since Kali Linux on Windows Terminal seems to render it properly

**Comment by ZePhant** [comment]
> The terminal might not be the problem @tombh , since Kali Linux on Windows Terminal seems to render it properly

**Comment by tombh** [comment]
There are a couple of tests that that gist mentions, I'd be interested to see their output on the problematic terminal

**Comment by zstg** [comment]
![image](https://user-images.githubusercontent.com/69384921/159463782-4a85de6d-d320-4d63-ac7b-eaca904e103a.png)
> MSYS Bash (Windows Terminal)

Guess this means Windows Terminal supports TrueColour

**Comment by tombh** [comment]
Yes, that's right. But do you have the same problem as @ZePhant?

**Comment by zstg** [comment]
Yeah

Colours are indeed messed up on Windows Terminal

**Comment by tombh** [comment]
Wait, I'm confused. The test above was in Windows Terminal, which means that it's supporting true colour. But when you run Browsh in that same terminal you get the same ugly banding artefacts?

**Comment by zstg** [comment]
Yes

It renders properly on Linux , but poorly on Windows (cmd/PowerShell/MSYS Bash)

**Comment by tombh** [comment]
Right, so to be specific: those true colour tests you posted work for the Windows Terminal in _both_ Linux and Windows? The problem is that even though Windows Terminal passes the true colour tests on Windows it cannot render Browsh correctly on Windows?

**Comment by zstg** [comment]
Exactly

Both run on the same Windows Terminal

**Comment by zstg** [comment]
To confirm, I have Windows Terminal on Windows *only* (duh) and i run both cmd/ps/GitBash *and* my WSL distros from there

**Comment by zstg** [comment]
> Right, so to be specific: those true colour tests you posted work for the Windows Terminal in _both_ Linux and Windows? The problem is that even though Windows Terminal passes the true colour tests on Windows it cannot render Browsh correctly on Windows?

Yes , I’m using Windows Terminal with a profile for PowerShell  and one for WSL

The WSL version of browsh is flawless but the Windows version on the same Terminal is not able to render colours correctly

Sorry if i was a bit unclear in the previous message

**Comment by tombh** [comment]
I'm sorry, I'm still not clear 🥺 Maybe it's that I haven't used Windows in decades, I'm not very familiar with all the terms and tech.

Basically, what I'm interested in here is just if a terminal can render TrueColor but not render Browsh properly. That would mean that Browsh has a bug. Any other case it's likely a problem with the terminal itself.

**Comment by Srinath-Anand** [comment]
Windows Terminal supports profiles. WSL is a way ( ~ a VM)  to run Linux distros on Windows. Windows Terminal on Windows supports these WSL profiles. 
This allows users to run bash from Windows Terminal

**Comment by zstg** [comment]
> I'm sorry, I'm still not clear 🥺 Maybe it's that I haven't used Windows in decades, I'm not very familiar with all the terms and tech.
> 
> Basically, what I'm interested in here is just if a terminal can render TrueColor but not render Browsh properly. That would mean that Browsh has a bug. Any other case it's likely a problem with the terminal itself.

The same(on the one OS) TrueColor terminal renders properly in WSL and poorly in Windows

**Comment by tombh** [comment]
So if TrueColor renders poorly in Windows then the bug isn't in Browsh right? You understand that? TrueColor is a requirement for Browsh

**Comment by zstg** [comment]
Is that how it works on the **same** terminal ? Windows Terminal **can and does** render TrueColor …


TrueColor works perfectly fine on both Windows and WSL(Linux)

**Comment by tombh** [comment]
I'm just more and more confused 😢

Your screenshots indeed show that TrueColor is working in whatever terminals they are where ever they are running. I see the word "Arch Linux", so does that mean it's real Linux, or a VM on Windows? Or WSL? I'm so lost 🥺 And why does one terminal have a different coloured background? Is that 2 tmux panes? Or 2 terminals?

**Comment by zstg** [comment]
WSL *lets* you run Linux on Windows 

I just changed the background colour  for the 2 profiles - they’re 2 separate panes running in a split window

**Comment by lofcz** [comment]
This is indeed a bug in Browsh. Windows Terminal supports true color and it works correctly in it when running Browsh under WSL/2.

**Comment by tombh** [comment]
Thanks for the feedback. Can you answer any of the questions in my previous comment?

**Comment by zstg** [comment]
> I'm just more and more confused 😢
> 
> Your screenshots indeed show that TrueColor is working in whatever terminals they are where ever they are running. I see the word "Arch Linux", so does that mean it's real Linux, or a VM on Windows? Or WSL? I'm so lost 🥺 And why does one terminal have a different coloured background? Is that 2 tmux panes? Or 2 terminals?

Both windows are running side-by-side in a Windows Terminal split pane

**Comment by lofcz** [comment]
> I see the word "Arch Linux", so does that mean it's real Linux, or a VM on Windows? Or WSL? I'm so lost 

WSL

> And why does one terminal have a different coloured background?

It is irrelevant to the problem but @zstg has changed the background for both panes as per his comment

> Is that 2 tmux panes? Or 2 terminals?

2 Windows Terminal panes

Does this help?

**Comment by tombh** [comment]
Great, thanks, yes that helps. So that screenshot is 2 Windows Terminals panes, the one on the left is just BASH running normally under Windows, and the one on the right is Arch Linux running under WSL. They both clearly support true colour. Ideally what I'd like to see is those exact same panes after `docker run browsh/browsh` has been run in each of them.

**Comment by zstg** [comment]
> > I see the word "Arch Linux", so does that mean it's real Linux, or a VM on Windows? Or WSL? I'm so lost
> 
> WSL
> 
> > And why does one terminal have a different coloured background?
> 
> It is irrelevant to the problem but @zstg has changed the background for both panes as per his comment
> 
> > Is that 2 tmux panes? Or 2 terminals?
> 
> 2 Windows Terminal panes
> 
> Does this help?
Thanks for answering , been quite busy lately !!

**Comment by zstg** [comment]
Oh .. so it's not the fact that TrueColour renders properly on WSL ..?

No probs, I'll stick to the WSL version ;)

Thanks a lot !
